### PR TITLE
Handle tabs after backslash at end of file

### PR DESCRIPTION
**Pull request by dbrotz**
Recently, rgbasm was modified to support tabs between a backslash and newline it escapes, but `yy_create_buffer()` was not updated to handle tabs. This meant that a backlash followed by a tab at the end of the file could crash rgbasm. I updated `yy_create_buffer()` and added a test for trailing whitespace after a backlash at the end of the file.

It seems that `make checkpatch` doesn't like trailing whitespace even in test files, so I expect that check will fail when the CI runs.

### CODE REVIEW

**Comment by ISSOtm** [comment]
I'm looking for a way to have checkpatch ignore `.asm` files...

**Comment by dbrotz** [comment]
It doesn't seem like there is a (simple) way to do so. I don't see why it should hold back this PR in any case.

I found more buffer overflows of a similar nature in macro and rept blocks, so I added another commit to fix those as well.

**Comment by dbrotz** [comment]
I forgot to add those to the commit. I've added them now.

File: src/asm/lexer.c [hunk 1]
```diff
@@ -167,24 +167,47 @@ static void yy_buffer_append(YY_BUFFER_STATE buf, uint32_t capacity, char c)
...
 YY_BUFFER_STATE yy_scan_bytes(char *mem, uint32_t size)
 {
 	YY_BUFFER_STATE pBuffer = malloc(sizeof(struct yy_buffer_state));
 
 	if (pBuffer == NULL)
 		fatalerror("%s: Out of memory!", __func__);
 
-	pBuffer->pBufferRealStart = malloc(size + 1 + SAFETYMARGIN);
+	size_t capacity = size + 3; /* space for 2 newlines and terminator */
+	pBuffer->pBufferRealStart = malloc(capacity + SAFETYMARGIN);
```
Comment (ISSOtm) [inline src/asm/lexer.c]: Can you fix [the warning this causes](https://travis-ci.org/rednex/rgbds/jobs/583086153#L239)?

Comment (dbrotz) [reply src/asm/lexer.c]: I personally disagree with some of these coding style rules that `checkpatch` enforces... I don't mind changing it, though.

Comment (ISSOtm) [reply src/asm/lexer.c]: Me neither, but it's there.

**Comment by ISSOtm** [comment]
Checkpatch will fail on this precise commit, but that won't matter as it's just for one commit; `make checkcodebase` will probably fail, but `.asm` files can be ignored from its input set. Also, `make checkcodebase` is already broken as-is, so it'll just be another one on the pile...

**Pull request merged**